src/librabbitmq/amqp_private.h

```diff
@@ -125,6 +125,14 @@ extern int amqp_encode_table(amqp_bytes_t encoded,
     _result;					\
   })
 
+#define AMQP_CHECK_WRITE_RESULT(expr)			\
+  ({						\
+    int _result;            \
+    do {_result = (expr);} while (_result == -EINTR);			\
```

Comment:
write(2) will return -1 on error and set errno to EINTR...

I believe that should be

```
while(-1 == (_result = expr)) && errno == EINTR);
```